staging/src/k8s.io/apimachinery/pkg/apis/meta/v1beta1/conversion.go

```diff
@@ -16,12 +16,31 @@ limitations under the License.
 
 package v1beta1
 
-import "k8s.io/apimachinery/pkg/conversion"
+import (
+	"unsafe"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+)
 
 // Convert_Slice_string_To_v1beta1_IncludeObjectPolicy allows converting a URL query parameter value
-func Convert_Slice_string_To_v1beta1_IncludeObjectPolicy(input *[]string, out *IncludeObjectPolicy, s conversion.Scope) error {
-	if len(*input) > 0 {
-		*out = IncludeObjectPolicy((*input)[0])
+func Convert_Slice_string_To_v1beta1_IncludeObjectPolicy(in *[]string, out *IncludeObjectPolicy, s conversion.Scope) error {
+	if len(*in) > 0 {
+		*out = IncludeObjectPolicy((*in)[0])
 	}
 	return nil
 }
+
+// Convert_v1beta1_PartialObjectMetadataList_To_v1_PartialObjectMetadataList allows converting PartialObjectMetadataList between versions
+func Convert_v1beta1_PartialObjectMetadataList_To_v1_PartialObjectMetadataList(in *PartialObjectMetadataList, out *v1.PartialObjectMetadataList, s conversion.Scope) error {
+	out.ListMeta = in.ListMeta
```

Comment:
what did the reflective conversion do with TypeMeta? did it ignore it or preserve it?

Reply:
I guess you ask about ListMeta? For ListMeta, it's being copied.

That said, for TypeMeta, unless I misunderstand something, we have a bit of a mess.
- auto-generated conversions are ignoring it
- there is also manual conversion to ignore it, but it's NOT registered anywhere:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go#L132
- default converter is copying it (because unless there isn't a registered conversion, it's copying everything)
- convertToVersion is setting it:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go#L531
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go#L124
So I guess we need to clean it up at some point (though I'm not sure what is the right thing to do here...)

Reply:
I was asking about TypeMeta, just happened to click on this line. Can you verify that getting PartialMetadataList in v1 and v1beta1 gets returned with the correct kind/apiVersion?

Reply:
stuff like curl -H "Accept: application/json;as=PartialObjectMetadataList;g=meta.k8s.io;v=v1beta1" localhost:8080/api/v1/pods have TypeMeta set (both with my PR and without)

It seems to be because ConvertToVersion is used to serve those and it sets it explicitly:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go#L531

Reply:
thanks for checking